src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.CoreCLR.cs

```diff
@@ -84,6 +94,34 @@ Signature LazyCreateSignature()
         private RuntimeType ReflectedTypeInternal => m_reflectedTypeCache.GetRuntimeType();
 
         internal BindingFlags BindingFlags => m_bindingFlags;
+
+
+        [DebuggerStepThrough]
+        [DebuggerHidden]
+        internal unsafe object InvokeNonEmitUnsafe(object? obj, IntPtr* args, Span<object?> argsForTemporaryMonoSupport, BindingFlags invokeAttr)
+        {
+            if ((invokeAttr & BindingFlags.DoNotWrapExceptions) == 0)
```

Comment:
Handling of the `DoNotWrapExceptions` flag should be in the static code that is shared between strategies. I do not think there is anything to gain by handling it in ref-emited code.

Reply:
(Maybe I do not see how you plan to add the ref-emit path.)

Reply:
It ended up this way because Mono has different exception logic, and the invoker code is shared with Mono.

Reply:
So if we end up having the same exception handling then yes this exception handling will be moved to the common invoker location.